File: Lab_03/appoo3/appoo3/CustomDeque.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace appoo3
{
    class CustomDeque
    {
        private LinkedList<SmartPhone> dq;

        public CustomDeque()
        {
            this.dq = new LinkedList<SmartPhone>();
        }

        public void AddElement(SmartPhone obj)
        {
            this.dq.AddLast(obj);
        }

        public LinkedList<SmartPhone> GetDeque()
        {
            return dq;
        }

        public void Print()
        {
            Console.WriteLine("\n[Print Deque]\n");
            foreach(var obj in dq)
            {
                Console.WriteLine(obj);
            }
        }

        public void SortAsceding()
        {
            CustomList temp = new CustomList();
            foreach(var obj in dq)
            {
                temp.AddElement(obj);
            }
            temp.SortAscending();
            dq.Clear();
            foreach(var obj in temp.GetList())
            {
                dq.AddLast(obj);
            }
        }
    }
}

File: Lab_02/APPOO2_3/src/com/company/CustomDeque.java
package com.company;

/**
 * Created by bossa on 13.06.2017.
 */
import java.util.*;

class CustomDeque{
    Deque<SmartPhone> dq;

    CustomDeque(){
        this.dq = new ArrayDeque<>();
    }

    void addElement(SmartPhone p)
    {
        dq.add(p);
    }

    void sortAscending(){
        CustomList temp = new CustomList();
        for(SmartPhone obj : dq) temp.addElement(obj);
        temp.sortAscending();
        temp.writeToDeque(dq);
    }

    void PrintDeque(){
        System.out.println("Print Deque: ");
        for(SmartPhone obj : dq){
            System.out.println(obj);
        }
    }

    CustomList AddToList(CustomList list){
        CustomList result = new CustomList();
        for(SmartPhone obj : list.getList()){
            result.addElement(obj);
        }
        for(SmartPhone obj : dq){
            result.addElement(obj);
        }
        return result;
    }
}